OOP-Milestone:User interface

IjayAbby opened the pull request:
In this project, we are creating a user-interface of the Game Logic while using 'puts' and 'get.chomps' commands.
We have created two players with "X" and "0" making sure our main file is executable.

Div685 committed:
Add bin and lib Folder

- Add main file under bin folder
- Add .gitkeep file

IjayAbby committed:
rename main.rb to main

Div685 committed:
update Readme.md file

- update Gemfile for rubocop linter

Div685 committed:
update Rspec Linter

IjayAbby committed:
update author details
license
typing errors

IjayAbby committed:
add license link

IjayAbby committed:
add lib folder and move .gitkeep file inside it

Div685 committed:
Add Player method

- Add player X and print statement

IjayAbby committed:
add player2

IjayAbby committed:
add player to make file executable

IjayAbby committed:
add final newline

mimipeshy commented:
## STATUS: CHANGES REQUIRED :red_circle: 

Hey @IjayAbby  and @Div685  :wave:  :wave: , you guys did an impressive job with this milestone, you are almost done but let's make a few changes first. :smile: 

**Before we start its important to note that:-** 
-  In this step students are supposed to only **mock the basic flow of the game**. This means **implementing a user interface** that displays prompt for users (print/puts) and collects user input (get.chomp). No game logic (for example checking who is a winner) is required in this milestone - instead, students can just display "The winner is X" no matter who won. Usage of inline comments in this milestone is allowed. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

**What we need to add to the User Interface**
##### Behaviour
- [x] `[REQUIRED]` game informs player if the selected move is invalid (without computing what an invalid move is)

- [x] `[REQUIRED]` game displays board after the player move (without computing which spots on the board are taken)

- [x] `[REQUIRED]` game informs player if the selected move is a winning move (without computing the winning move)

- [x] `[REQUIRED]` game informs player if the selected move is a draw move (without computing draw)

- [x] `[REQUIRED]` game repeats all actions for the next player's move

##### Implementation details

* [x] A loop is needed in this game (recommended is `while` but anything else will do) ([OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png))
* [x] Inside the loop, the student should call actions repeated for each move
* [x] A loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with variable:
* [x] Using out-of-nowhere numbers to terminate a loop, for example, `5.times do ... end` is a signal that something is wrong. Advise usage of terminating variable as below.
```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

That's all :smile:  , just work on the above and submit another form for review
 Keep up the good work :muscle:  :smile: 

If you need further clarification dont hesitate to contact me on slack (peris ndanu)

Regards,
Peris Ndanu

Div685 committed:
Add methods in main class

- Defined methods and comments for descriptions

Div685 committed:
update main and board file

- resolve rubocop worning
- display board with number in board file

jubaan reviewed (changes requested):
# Status: Changes Required :recycle: 

## :wave: :wave: Hello @ . Thanks for submitting you project!

The code in the `bin/main.rb` file corresponds to the logic, those are the classes and methods that will let the players play and enjoy the game, while they can be related to the interface of the game they are far apart from being the interface itself.

The objective of this milestone is to create the **User Interface**, or in other words, the workflow the game will follow to apply the logic or the code that the `bin/main.rb` file has right now.

To make it more understandable, the **User interface** is the decision path the player and the game is going to follow. for example:

```Ruby

# Introduction to the game
puts "Welcome players this is TicTacTronix3000, a game based on the normal tic tac toe game"

# Ask for players name
puts "Player one please write your name"
player1 = gets.chomp

puts "Player two please write your name"
player2 = gets.chomp

# Start game

... some more code ...

```

There are some issues that need to be addressed before we can move forward. Please review the list below.

## Required changes :wrench: 
### [Mandatory] :red_circle: 

- [x] Read the description and examples above.

- [x] Please use the information and the previous TSE recommendations to create the user interface.

## Keep up the great work :muscle: and Happy Coding :smiley: :computer:!!

Div685 committed:
Update main file for user interface

- Removed logic parts of class
- Add loop
- Add winning player
- Add different statement in loop

ElijahTobs reviewed (changes requested):
# Status: Changes Required :recycle: 

### Hi @IjayAbby and @Div685 

You two have tried but you need to make some more changes before this milestone can be approved.

I tried to play the game and I noticed that after supplying the names of player 1 and 2, player 2 was asked to play. I made a valid move and nothing happened. I later made the same move but this time, your game tells me I've won and also says it's a draw.


## [REQUIRED]

- [x] game informs player if the selected move is invalid (without computing what an invalid move is)
- [x] game displays board after the player move (without computing which spots on the board are taken)
- [x] game informs player if the selected move is a winning move (without computing the winning move)
- [x] game informs player if the selected move is a draw move (without computing draw)
- [x] game repeats all actions for the next player's move

## Kindly make the required changes and submit for re-review

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

Div685 committed:
update main file

- Add Win move
- Add loop for player_turn

descholar-ceo commented on bin/main:
```diff
@@ -0,0 +1,114 @@
+        puts "Valid moves! " 
+        display_board(board.board)
+
+        (i % 2 == 0) ? (puts "Player1 Your turn") : (puts "Player2 Your Turn")
```
Use the real player names, here!

descholar-ceo reviewed (changes requested):
# Status: Change Required :vertical_traffic_light: :warning: 
## Hello @IjayAbby and @Div685  :handshake: :handshake: :handshake: 
:+1: This is good to make it this far, this milestone is almost done, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
:+1: Congratulations on adding readme file at this stage, that is very good!
### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
## Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [x] Remove all `.gitkeep` files from your folders, you no longer need them
- [x] The game should use the real inputed names when calling the player's turn
- [x] Implement all changes I suggested in inline comments in your codes

# [Optional changes] :yellow_circle: 
- [x] Remember to move all `puts or gets` statements to `main` file, the ones which are in `lib/board.rb` file are not recommended

Re-reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)
_slack:_ `@Mugirase Emmanuel` | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

Div685 committed:
update valid_input method

- Add name1 and name1 in valid_input method

IjayAbby committed:
Add inline comments on methods

umairarshadbutt reviewed (approved):
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.**
- Congratulations on the splendid job @IjayAbby and @Div685! :+1: 
- You can now merge to master and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal:
##
**If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)
